### PR TITLE
allow direct pushing of certs to GitHub T7582

### PR DESCRIPTION
**Pull request by Reception123**

### CODE REVIEW

File: modules/ssl/files/ssl-certificate.py [hunk 1]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
 
         os.system(f'/bin/cat /etc/letsencrypt/live/{self.domain}/fullchain.pem')
 
+        os.system(f'git config --global core.sshCommand "ssh -i /var/lib/nagios/id_rsa -F /dev/null -o ProxyCommand=\'nc -6 -X connect -x bast.miraheze.org:8080 %h %p\'"')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:129]:
```suggestion
        os.system('git config --global core.sshCommand "ssh -i /var/lib/nagios/id_rsa -F /dev/null -o ProxyCommand=\'nc -6 -X connect -x bast.miraheze.org:8080 %h %p\'"')
```

File: modules/ssl/files/ssl-certificate.py [hunk 2]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'git -C /srv/ssl/ssl/ config user.email "noreply@miraheze.org" &&   git -C /srv/ssl/ssl/ config user.name "MirahezeSSLBot"')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:130]:
```suggestion
        os.system('git -C /srv/ssl/ssl/ config user.email "noreply@miraheze.org" &&   git -C /srv/ssl/ssl/ config user.name "MirahezeSSLBot"')
```

File: modules/ssl/files/ssl-certificate.py [hunk 3]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'cp /etc/letsencrypt/live/{self.domain}/fullchain.pem /srv/ssl/ssl/certificates/{self.domain}.crt')
+        os.system(f'git -C /srv/ssl/ssl/ add /srv/ssl/ssl/certificates/{self.domain}.crt')
+        os.system(f'echo \'{self.domain}:\' >> /srv/ssl/ssl/certs.yaml')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:133]:
```suggestion
        os.system(f"echo '{self.domain}:' >> /srv/ssl/ssl/certs.yaml")
```

File: modules/ssl/files/ssl-certificate.py [hunk 4]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'echo \'\t url: \'{self.domain}\' \' >> /srv/ssl/ssl/certs.yaml')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:134]:
```suggestion
        os.system(f"echo '\t url: '{self.domain}' ' >> /srv/ssl/ssl/certs.yaml")
```

File: modules/ssl/files/ssl-certificate.py [hunk 5]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'echo \'\t ca: \'LetsEncrypt\' \' >> /srv/ssl/ssl/certs.yaml')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:135]:
```suggestion
        os.system("echo '\t ca: 'LetsEncrypt' ' >> /srv/ssl/ssl/certs.yaml")
```

File: modules/ssl/files/ssl-certificate.py [hunk 6]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'echo \'\t disable-event: \'false\' \n \' >> /home/reception/certs.yaml')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:136]:
```suggestion
        os.system("echo '\t disable-event: 'false' \n ' >> /home/reception/certs.yaml")
```

File: modules/ssl/files/ssl-certificate.py [hunk 7]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'git -C /srv/ssl/ssl/ add /srv/ssl/ssl/certs.yaml')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:137]:
```suggestion
        os.system('git -C /srv/ssl/ssl/ add /srv/ssl/ssl/certs.yaml')
```

File: modules/ssl/files/ssl-certificate.py [hunk 8]
```diff
@@ -126,6 +126,18 @@ def generate_letsencrypt_certificate(self):
...
+        os.system(f'git -C /srv/ssl/ssl/ commit -m "Bot: Add SSL cert for {self.domain}"')
+        os.system(f'git -C /srv/ssl/ssl/ push origin master')
```
Comment (MacFan4000) [inline modules/ssl/files/ssl-certificate.py:139]:
```suggestion
        os.system('git -C /srv/ssl/ssl/ push origin master')
```

**Review by MacFan4000 — CHANGES_REQUESTED** [review]
These suggestions fix the broken CI

**Comment by Reception123** [comment]
After discussing with @Universal-Omega I'll leave it to him to correct the issues as he wants to make some other improvements as well.

File: modules/ssl/files/ssl-certificate.py (hunk 1 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 3 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 5 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 7 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 8 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 6 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 4 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

File: modules/ssl/files/ssl-certificate.py (hunk 2 above)
Comment (Universal-Omega) [reply modules/ssl/files/ssl-certificate.py]: Done.

**Comment by Reception123** [comment]
The new changes replace everything in certs.yaml with the new wiki (so removes all the contents) - see https://github.com/miraheze/ssl/commit/cd54c7cdcb9e410aae42325ab55515a993da8148

**Comment by Universal-Omega** [comment]
> The new changes replace everything in certs.yaml with the new wiki (so removes all the contents) - see [miraheze/ssl@cd54c7c](https://github.com/miraheze/ssl/commit/cd54c7cdcb9e410aae42325ab55515a993da8148)

That is fixed now.

**Review by Universal-Omega — APPROVED** [review]

**Pull request merged**